src/main/webapp/script.js

```diff
@@ -36,3 +41,14 @@ function checkLogin() {
     document.getElementById('login-tab').href = status.changeLogInStatusURL;
   });
 }
+
+// If user is not registered, show the registration message.
+function ifUnregisteredShowRegistrationMessage() {
+  fetch('/person')
+    .then(response => {
+      // response.redirected means they are not registered yet.
```

Comment:
This is a hack. You're making the request for person information, but not to get that information, but just to know if the backend wants to redirect us.

Is `status.loggedIn` (in checkLogin) equivalent?

Reply:
No, this is to check if they are registered, not if they are logged in.

Reply:
How are you going to check that they are registered if they are not logged in? If they're logged in then they are registered.

Does "registered" mean something different from "made an account"?

Reply:
They log in with their google account (because we're using google's api) and then we ask them to make an account for our site by registering. So registering is making an account, but you do not need an account for our site to log in.

Reply:
That should be fixed. We should not require a separate step from oauth (that's bad UX).